### PR TITLE
Create a markdown rule, rather than injecting the parser

### PR DESCRIPTION
**Pull request by Vap0r1ze**
This works with 12944qwerty/chat-components#1

> Worth noting this solution will revert previous injections to `parser.parse`
> I think that's okay, anything injecting into `parser.parse` create a markdown rule instead

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,27 +4,50 @@ const { inject, uninject } = require('powercord/injector');
...
+				if (match === null) return null;
+				const desc = getIndicator(match[1]);
+				if (desc === null) return null;
+				return match;
+			},
+			parse: match => ({
+				indicator: match[1],
+				desc: getIndicator(match[1]),
+			}),
+			react: node => ToneIndicator(node),
```
Comment (asportnoy) [inline index.js]: Could you do this shorthand?
```suggestion
			react: ToneIndicator,
```

File: index.js [hunk 2]
```diff
@@ -4,27 +4,50 @@ const { inject, uninject } = require('powercord/injector');
...
-		const parser = await getModule(['parse', 'parseTopic']);
-		const process = this.process.bind(this);
+		const parser = this.parser = await getModule(['parse', 'parseTopic']);
+
+		await this.createEmbedTitleRules();
+
+		parser.defaultRules.toneIndicator = {
+			order: parser.defaultRules.text.order - 1,
+			match: (source, state) => {
+				if (state.prevCapture && !LOOKBEHIND_PATTERN.test(state.prevCapture)) {
```
Comment (asportnoy) [inline index.js]: Is `prevCapture` controlled by Discord?

**Review by asportnoy — CHANGES_REQUESTED** [review]
Some regex not working properly, see Discord for info

File: index.js (hunk 2 above)
Comment (Vap0r1ze) [reply index.js]: nah see https://github.com/Khan/simple-markdown#extension-overview

**Review by asportnoy — APPROVED** [review]

**Pull request merged**